pkg/platform/kube/functionres/lazy.go

```diff
@@ -279,64 +279,76 @@ func (lc *lazyClient) WaitAvailable(ctx context.Context,
 			return err, functionconfig.FunctionStateUnhealthy
 		}
 
-		// get the deployment. if it doesn't exist yet, retry a bit later
-		result, err := lc.kubeClientSet.AppsV1().
-			Deployments(namespace).
-			Get(ctx, deploymentName, metav1.GetOptions{})
-		if err != nil {
-			continue
+		if deploymentReady && ingressReady {
+			return nil, functionconfig.FunctionStateReady
 		}
 
-		// find the condition whose type is Available - that's the one we want to examine
-		for _, deploymentCondition := range result.Status.Conditions {
+		// deployment is ready, ingress is not yet (being too slow I guess, marking as unhealthy)
+		if deploymentReady && !ingressReady && time.Since(timeDeploymentReady) >= time.Minute {
```

Comment:
Why is 1 min the "Unhealthy" threshold?
What if `waitReadinessTimeoutBeforeFailure` is set to true?